modules/redis/options.go

```diff
@@ -0,0 +1,129 @@
+	"crypto/tls"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/mdelapenya/tlscert"
+
+	"github.com/testcontainers/testcontainers-go"
+)
+
+type options struct {
+	tlsEnabled bool
+	tlsConfig  *tls.Config
+}
+
+// Compiler check to ensure that Option implements the testcontainers.ContainerCustomizer interface.
+var _ testcontainers.ContainerCustomizer = (Option)(nil)
+
+// Option is an option for the Redpanda container.
+type Option func(*options) error
+
+// Customize is a NOOP. It's defined to satisfy the testcontainers.ContainerCustomizer interface.
+func (o Option) Customize(*testcontainers.GenericContainerRequest) error {
+	// NOOP to satisfy interface.
+	return nil
+}
+
+// WithTLS sets the TLS configuration for the redis container, setting
+// the 6380/tcp port to listen on for TLS connections and using a secure URL (rediss://).
+func WithTLS() Option {
+	return func(o *options) error {
+		o.tlsEnabled = true
+		return nil
+	}
+}
+
+// WithConfigFile sets the config file to be used for the redis container, and sets the command to run the redis server
+// using the passed config file
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	const defaultConfigFile = "/usr/local/redis.conf"
+
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
+			ContainerFilePath: defaultConfigFile,
+			FileMode:          0o755,
+		}
+		req.Files = append(req.Files, cf)
+
+		// prepend the command to run the redis server with the config file, which must be the first argument of the redis server process
+		req.Cmd = append([]string{defaultConfigFile}, req.Cmd...)
+
+		return nil
+	}
+}
+
+// WithLogLevel sets the log level for the redis server process
+// See "[RedisModule_Log]" for more information.
+//
+// [RedisModule_Log]: https://redis.io/docs/reference/modules/modules-api-ref/#redismodule_log
+func WithLogLevel(level LogLevel) testcontainers.CustomizeRequestOption {
+	return testcontainers.WithCmdArgs("--loglevel", string(level))
+}
+
+// WithSnapshotting sets the snapshotting configuration for the redis server process. You can configure Redis to have it
+// save the dataset every N seconds if there are at least M changes in the dataset.
+// This method allows Redis to benefit from copy-on-write semantics.
+// See [Snapshotting] for more information.
+//
+// [Snapshotting]: https://redis.io/docs/management/persistence/#snapshotting
+func WithSnapshotting(seconds int, changedKeys int) testcontainers.CustomizeRequestOption {
+	if changedKeys < 1 {
+		changedKeys = 1
+	}
+	if seconds < 1 {
+		seconds = 1
+	}
+
+	return testcontainers.WithCmdArgs("--save", strconv.Itoa(seconds), strconv.Itoa(changedKeys))
+}
+
+// createTLSCerts creates a CA certificate, a client certificate and a Redis certificate,
+// storing them in the given temporary directory.
+func createTLSCerts() (*tlscert.Certificate, *tlscert.Certificate, *tlscert.Certificate, error) {
```

Comment:
suggestion: as all three types are the same I would name the return values so the consumer knows what to expect without having to read the function. Also as the consumer only needs the bytes you could just return that from this function.

Reply:
they need the bytes and access to the TLSConfig, that's why I returned the entire type.

Will update the signature to name the returned elements 👍

Reply:
thanks for clarifying, missed that.